Update

XedinUnknown opened the pull request:
- Drop PHP < 7.4;
- Upgrade PHPUnit;
- Fix and upgrade Psalm;
- QoL improvements;
- Minor fixes.

XedinUnknown committed:
Update config

XedinUnknown committed:
Allow plugin and update deps

XedinUnknown committed:
Format one version per line

XedinUnknown committed:
Fix variable not propagating during total build

XedinUnknown committed:
Add `XDEBUG_VERSION` env var

XedinUnknown committed:
Drop PHP < 7.4

Now also testing on newer versions.
Updated deps and configs.

XedinUnknown committed:
Upgrade Psalm and fix its config

XedinUnknown committed:
Update config

XedinUnknown committed:
Update deps

strangerkir commented on .env.example:
```diff
@@ -2,7 +2,8 @@ BASE_PATH=./
-PHP_BUILD_VERSION=7.1
-PHP_TEST_VERSION=7.1
+PHP_BUILD_VERSION=7.4
+PHP_TEST_VERSION=8.2
```
Is PHP `8.2` chosen intentionally for tests? Testing with the lowest possible PHP version is probably better decision.

XedinUnknown replied:
Yes, that is intentional. With regard to what version we test with, that really depends on what you mean by "test". The `.env` vars are used to set up the local environment, and `PHP_TEST_VERSION` is the version of PHP in the container, where you run your QA tools during development. Consequently, it means that PHPUnit will run on 8.2.

This project supports PHP `^7.4 | ^8.0`, meaning there are 4 PHP versions to test on. CI [runs on all of them](https://github.com/Dhii/containers/blob/update/.github/workflows/ci.yml#L11-L14), but locally you've got to choose one. Since the project is already built on PHP 7.4, meaning for example that its deps are installed on that version of PHP, and will likely be mostly used by newer PHP versions, it makes sense to test on the latest PHP version to ensure compatibility between the EoL version of the build and the current latest stable of PHP while developing. Once again, though: automation will test on more versions.

Does that address your concern? What are your thoughts?

strangerkir replied:
> Does that address your concern? What are your thoughts?

Yes, sure. I'm fine with this, just wanted to make sure this is not a mistake, but a decision. This logic completely makes sense to me, thanks for the explanation!

strangerkir reviewed (commented):
You may want to remove [deprecated](https://github.com/vimeo/psalm/issues/6892) `ForbiddenEcho` from Psalm config.
Also, not sure about the default PHP test version.
The rest seems fine to me.

XedinUnknown commented:
Heya @strangerkir! Thanks for pitching in!

Yes, Psalm has things to fix. Unfortunately, there are many Psalm-reported issues that I'm not sure why they are being reported: it's like it's failing to infer types from PHPDoc, and errs because e.g. parent interfaces don't have return types declared (PSR). I'm going to be looking more at this soon.

XedinUnknown committed:
Remove deprecated config

XedinUnknown committed:
Add type information

XedinUnknown committed:
Merge remote-tracking branch 'origin/0.2.x' into update

# Conflicts:
#	.idea/php.xml
#	composer.lock

XedinUnknown committed:
Fix tests and docs; update deps and configs

XedinUnknown committed:
Fix more tests and docs

XedinUnknown committed:
Disallow PHPUnit 10

The many BC breaks and removed features in that version make it not worth upgrading this package to at this time.
